src/sample.py

```diff
@@ -39,7 +40,7 @@ def message (self, buf, font, msg, data, _x_what_is_this):
 		return True
 debugger = Debugger()
 hb.buffer_set_message_func (buf, debugger.message, 1, 0)
-hb.buffer_add_utf8 (buf, text.encode('utf-8'), 0, -1)
+hb.buffer_add_codepoints (buf, codepoints, 0, -1)
```

Comment:
if add_utf32() does everything add_codepoints() does and better, why are
there two functions?

On Fri, Jun 17, 2016 at 7:09 PM, Khaled Hosny <notifications@github.com>
wrote:

> In src/sample.py
> <https://github.com/behdad/harfbuzz/pull/269#discussion_r67586781>:
>
> > @@ -39,7 +40,7 @@ def message (self, buf, font, msg, data, _x_what_is_this):
> >  		return True
> >  debugger = Debugger()
> >  hb.buffer_set_message_func (buf, debugger.message, 1, 0)
> > -hb.buffer_add_utf8 (buf, text.encode('utf-8'), 0, -1)
> > +hb.buffer_add_codepoints (buf, codepoints, 0, -1)
>
> It should.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/behdad/harfbuzz/pull/269/files/0a9c03bc45e82c95377431576ddc92e8cd32ddb4#r67586781>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACcEOvaipSrr2_lGpxJRClCrgkES_Xq_ks5qMykdgaJpZM4I3_8R>
> .
>